aiographite/aiographitesend.py

```diff
@@ -0,0 +1,325 @@
+
+			@metric_dir_list: List of string
+			@timestamp: the type should be int
+
+			If you're very confident that the metric name is valid, then use <method: send_single_valid_data> instead.
+
+		"""		
+		valid_metric_name = self._to_graphite_valid_metric_name(metric_dir_list)
+		self.send_single_valid_metric(valid_metric_name, value, timestamp)
+
+
+
+	@asyncio.coroutine
+	async def send_metric_list(self, dataset:List[Tuple] , timestamp = None):
+		"""
+			@param: 
+				Support two kinds of dataset
+
+				1)	dataset = [(metric_dir_list1, value1), (metric_dir_list2, value2), ...] 
+
+				or 
+
+				2)	dataset = [(metric_dir_list1, value1, timestamp1), (metric_dir_list1, value2, timestamp2), ...]
+
+			If you're very confident that the metric name is valid, then use <method: send_valid_dataset_list> instead.
+
+		"""
+		if not dataset:
+			return 
+
+		if len(dataset[0]) == 2:
+			valid_dataset = [(self._to_graphite_valid_metric_name(metric_dir_list), value) for metric_dir_list, value in dataset]
+		else:
+			valid_dataset = [(self._to_graphite_valid_metric_name(metric_dir_list), value, timestamp) for metric_dir_list, value, timestamp in dataset]
+
+		self.send_valid_metric_list(valid_dataset, timestamp)
+
+
+
+	@asyncio.coroutine
+	async def send_single_valid_metric(self, metric: str, value: int, timestamp = None):
+		"""
+			@metric: String
+			@value: int
+			@timestamp: int
+			Send a single data(metric value timestamp) to graphite
+		"""
+		timestamp = int(timestamp or time.time())
+
+		# Generate message based on protocol
+		listOfMetricTuples = [self.protocol.data_formate_function(metric, value, timestamp)]
+		message = self.protocol.generate_message_function(listOfMetricTuples)
+
+		# Sending Data
+		await self._send_message(message)
+
+
+
+	@asyncio.coroutine
+	async def send_valid_metric_list(self, dataset: List[Tuple], timestamp = None):
+		"""
+			@param: 
+			Support two kinds of dataset
+				1)	dataset = [(metric1, value1), (metric2, value2), ...] 
+				or 
+				2)	dataset = [(metric1, value1, timestamp1), (metric2, value2, timestamp2), ...]
+		"""
+		timestamp = int(timestamp or time.time())
+
+		# Generate message based on protocol
+		message = self._generate_message_for_data_list(dataset, timestamp, self.protocol.data_formate_function, self.protocol.generate_message_function)
+
+		# Sending Data
+		await self._send_message(message)
+
+
+
+	@asyncio.coroutine
+	async def send_valid_metric_dict(self, dataset: Dict, timestamp = None):
+		"""
+			Send data to graphite server when incoming data is in 'dict' format
+			@param: dataset = {
+									metric1 : value1,      // type ( string: int )
+									metric2 : value2, 
+									...
+							  }
+
+			metric1 (metric2, ...) are valid metric name for Graphite
+		"""
+		self.send_valid_metric_list(dataset.items(), timestamp)
+
+
+
+	@asyncio.coroutine
+	async def disconnect(self):
+		"""
+			Close the TCP connection 
+		"""
+		try:
+			self._writer.close()
+		except AttributeError:
```

Comment:
you can catch multiple exceptions using a tuple:

```
except (AttributeError, Exception):
```
But in this case, you already have the 'finally' block. So you don't need the others.